app/src/controllers/bucketPermission.js

```diff
@@ -112,14 +159,36 @@ const controller = {
       const userArray = mixedQueryToArray(req.query.userId);
       const userIds = userArray ? userArray.map(id => addDashesToUuid(id)) : userArray;
       const permissions = mixedQueryToArray(req.query.permCode);
-      const response = await bucketPermissionService.removePermissions(req.params.bucketId, userIds, permissions);
-      res.status(200).json(response);
+
+      const currUserId = await userService.getCurrentUserId(
+        getCurrentIdentity(req.currentUser, SYSTEM_USER), SYSTEM_USER);
+      const currBucketId = addDashesToUuid(req.params.bucketId);
+
+      if (isTruthy(req.query.recursive)) {
+
+        const parentBucket = await bucketService.read(currBucketId);
+        const childBuckets = await this._getChildrenWithManagePerms(currBucketId, currUserId);
+
+        const allBuckets = [parentBucket, ...childBuckets];
+
+        const responses = await Promise.all(
+          allBuckets.map(b =>
+            // Only remove permissions from child buckets that currentUser can MANAGE
+            bucketPermissionService.removePermissions(b.bucketId, userIds, permissions)
+          )
+        );
+        res.status(200).json(responses);
```

Comment:
to keep the response in the same shape we can do this:
`res.status(201).json(responses.flat());`

note: when doing recursive the response array includes all the recursive new/deleted perm records.. which I think is what we want.